### DISCUSSION TITLE
ckb 0.113.0 (82871a3 2024-01-09)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#4297] [Announcements]
### Highlights

- Tweak SendBlocksProof message to support ckb2023.
- Fixed incorrect and inadequate checks of sync message.
- Addressed issues with orphan pool for long pending transactions.
- Tuned rocksdb bloom filter for better performance.
- Improved persistence softfork cache.

### Compatibility Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2023 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Downloads

| OS                | Arch  | Runtime Dependencies | Package                 | Sign                |
| ----------------- | ----- | -------------------- | ----------------------- | ------------------- |
| macOS             | x64   | macOS 10.15 or above | [zip][macos-zip]        | [PGP][macos-pgp]    |
| macOS  (Portable) | x64   | macOS 10.15 or above | [zip][macos-p-zip]      | [PGP][macos-p-pgp]  |
| macOS             | ARM64 | macOS 10.15 or above | [zip][macosa-zip]       | [PGP][macosa-pgp]   |
| macOS  (Portable) | ARM64 | macOS 10.15 or above | [zip][macosa-p-zip]     | [PGP][macosa-p-pgp] |
| Linux             | x64   | glibc, libstdc++     | [tarball][linux-zip]    | [PGP][linux-pgp]    |
| Linux  (Portable) | x64   | glibc, libstdc++     | [tarball][linux-p-zip]  | [PGP][linux-p-pgp]  |
| CentOS            | x64   | glibc, libstdc++     | [tarball][centos-zip]   | [PGP][centos-pgp]   |
| CentOS (Portable) | x64   | glibc, libstdc++     | [tarball][centos-p-zip] | [PGP][centos-p-pgp] |
| Windows           | x64   | VC++ Redistributable | [zip][windows-zip]      | [PGP][windows-pgp]  |
| Linux             | ARM64 | glibc, libstdc++     | [tarball][arm-zip]      | [PGP][arm-pgp]      |
| Docker            |       |                      | [dockerhub][]           |                     |

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

Try the portable variants if the binaries complains about illegal instructions.

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-apple-darwin.zip.asc
[macos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-apple-darwin-portable.zip
[macos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-apple-darwin-portable.zip.asc
[macosa-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_aarch64-apple-darwin.zip
[macosa-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_aarch64-apple-darwin.zip.asc
[macosa-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_aarch64-apple-darwin-portable.zip
[macosa-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_aarch64-apple-darwin-portable.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-linux-gnu.tar.gz.asc
[linux-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-linux-gnu-portable.tar.gz
[linux-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-linux-gnu-portable.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-centos-gnu.tar.gz.asc
[centos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-centos-gnu-portable.tar.gz
[centos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-unknown-centos-gnu-portable.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.113.0/ckb_v0.113.0_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

#### Features

-   #4185: Tweak `SendBlocksProof` message to support ckb2023 (@quake)

#### Bug Fixes

-   #4192: Incorrect and inadequate checks of sync message (@yangby-cryptape)
-   #4199: Fix orphan pool for long pending tx issues (@chenyukang)
-   #4221: Fix typos (@xiaolou86)
-   #4218: Fix VM version select (@driftluo)

    This is a breaking change: b:consensus.

    - During synchronization, 2019 version transactions should use the v0 version of the VM.
    - RPC should faithfully represent the epoch in which the 2021 hardfork is located(just VM relations).

-   #4255: No chain_root in 1st block of the MMR activated epoch (@yangby-cryptape)
-   #4258: Fix concurrency issue for RBF (@chenyukang)
-   #4270: Resolve empty filter related message bug (@quake)

#### Improvements

-   #4191: Tuning rocksdb bloom filter (@quake)
-   #4203: Persistence softfork cache (@zhangsoledad)
-   #4235: Move `util/launcher/src/migrate.rs` to an independent crate (@eval-exec)
-   #4236: Break `ckb-launcher` and `ckb-chain`'s cycle  dependency by moving `SharedPackage` and `SharedBuilder` from `ckb-launcher` to `ckb-shared` (@eval-exec)
-   #4170: Use jsonrpc-utils to replace jsonrpc (@chenyukang)
-   #4256: Add aarch64 docker image  for ckb (@chenyukang)
-   #4288: `get_pool_tx_detail_info`: Change `score_sortkey`'s type from `String` to `struct AncestorsScoreSortKey`

    This is a breaking change: b:rpc. The schema of the response has changed.

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.113.0'>ckb 0.113.0 (82871a3 2024-01-09)</a>.</em>

**Comment by joshyates1980** [comment]
I'm curious.....will there be any future updates where the migration command will be needed?

**Comment by joshyates1980** [comment]
I modify the ckb.toml file after every full node update. 

### **I delete this array:**

bootnodes = [
  # Hongkong, Asia
  "/ip4/16.163.82.218/tcp/8114/p2p/QmaZMemLXSsxKUrYNucjEbPxVX3rBKsGhWW2muWtWxUWyh",
  # Tokyo, Asia
  "/ip4/35.79.196.111/tcp/8114/p2p/QmYCRVonLfP18LSoz2WCHaXDorUYxuUMfhtcXK1TuZ1iwF",
  # Mumbai, Asia
  "/ip4/13.234.144.148/tcp/8114/p2p/QmbT7QimcrcD5k2znoJiWpxoESxang6z1Gy9wof1rT1LKR",
  # Seoul, Asia
  "/ip4/34.64.120.143/tcp/8114/p2p/QmejEJEbDcGGMp4D6WtftMMVLkR1ZuBfMgyLFDMJymkDt6",
  # Virginia, North America
  "/ip4/3.218.170.86/tcp/8114/p2p/QmShw2vtVt49wJagc1zGQXGS6LkQTcHxnEV3xs6y8MAmQN",
  # Los Angeles, North America
  "/ip4/35.236.107.161/tcp/8114/p2p/QmSRj57aa9sR2AiTvMyrEea8n1sEM1cDTrfb2VHVJxnGuu",
  # Texas, North America
  "/ip4/23.101.191.12/tcp/8114/p2p/QmexvXVDiRt2FBGptgK4gBJusWyyTEEaHeuCAa35EPNkZS",
  # Toronto, North America
  "/ip4/20.151.143.237/tcp/8114/p2p/QmNsGNQjYA6iP472bNnNE2GR31kCYBifhY1XcaUxRjZ1py",
  # Frankfurt, Europe
  "/ip4/52.59.155.249/tcp/8114/p2p/QmRHqhSGMGm5FtnkW8D6T83X7YwaiMAZXCXJJaKzQEo3rb",
  # London, Europe
  "/ip4/3.10.216.39/tcp/8114/p2p/QmagxSv7GNwKXQE7mi1iDjFHghjUpbqjBgqSot7PmMJqHA",
  # Paris, Europe
  "/ip4/13.37.172.80/tcp/8114/p2p/QmXJg4iKbQzMpLhX75RyDn89Mv7N2H8vLePBR7kgZf6hYk",
  # Warsaw, Europe
  "/ip4/34.118.49.255/tcp/8114/p2p/QmeCzzVmSAU5LNYAeXhdJj8TCq335aJMqUxcvZXERBWdgS",
  # Victoria, Oceania
  "/ip4/40.115.75.216/tcp/8114/p2p/QmW3P1WYtuz9hitqctKnRZua2deHXhNePNjvtc9Qjnwp4q",
  # Santiago, South America
  "/ip4/34.176.239.95/tcp/8114/p2p/QmQoWrmuFauCn3zZ2mYYKAciG9opTbjzC2wVEfWveZNDt8",
  # Capetown, Africa
  "/ip4/13.245.217.98/tcp/8114/p2p/Qmf4t1SzFhRWuGcFcgs7r4pXvkACsz3FcaBMcmMKQMMpn7"
]

### **and  replace with this array:**

bootnodes = [
  # Virginia, North America
  "/ip4/3.218.170.86/tcp/8114/p2p/QmShw2vtVt49wJagc1zGQXGS6LkQTcHxnEV3xs6y8MAmQN",
  # Los Angeles, North America
  "/ip4/35.236.107.161/tcp/8114/p2p/QmSRj57aa9sR2AiTvMyrEea8n1sEM1cDTrfb2VHVJxnGuu",
  # Texas, North America
  "/ip4/23.101.191.12/tcp/8114/p2p/QmexvXVDiRt2FBGptgK4gBJusWyyTEEaHeuCAa35EPNkZS",
  # Toronto, North America
  "/ip4/20.151.143.237/tcp/8114/p2p/QmNsGNQjYA6iP472bNnNE2GR31kCYBifhY1XcaUxRjZ1py",
  # Probe Node
  "/ip4/52.33.75.45/tcp/8124/p2p/QmPHzhEzekyjB14eQsnsVRxmymtYmAyntcfJgVYRiUq4Lb"
]


**I also update max_outbound_peers = 20**

I make these updates in order to appear on: [https://nodes.ckb.dev/](https://nodes.ckb.dev/)
